shared/util.go

```diff
@@ -257,18 +257,36 @@ func LogPath(path ...string) string {
 	return filepath.Join(items...)
 }
 
-func ParseLXDFileHeaders(headers http.Header) (uid int64, gid int64, mode int, type_ string, write string) {
+// LXDFileHeaders is extracted from the `X-LXD-*` family of file permissions
+// headers.
+type LXDFileHeaders struct {
+	UID               int64
+	ModifyExistingUID bool
+
+	GID               int64
+	ModifyExistingGID bool
+
+	Mode               int
+	ModifyExistingMode bool
+
+	Type  string
+	Write string
+}
+
+// ParseLXDFileHeaders parses and validates the `X-LXD-*` family of file
+// permissions headers.
+func ParseLXDFileHeaders(headers http.Header) (LXDFileHeaders, error) {
```

Comment:
It would be really useful to the reader for the doc block of this function to document the format of the headers its going to parse.